adal/src/main/java/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -304,22 +304,30 @@ public String decrypt(final String encryptedBlob)
      */
     synchronized SecretKey loadSecretKeyForEncryption() throws IOException,
             GeneralSecurityException {
+        final byte[] secretKeyData = AuthenticationSettings.INSTANCE.getSecretKeyData();
+        return loadSecretKeyForEncryption(secretKeyData == null ? VERSION_ANDROID_KEY_STORE : VERSION_USER_DEFINED);
+    }
+
+    /**
+     * Get Secret Key based on API level to use in encryption. Decryption key
+     * depends on version# since user can migrate to new Android.OS
+     *
+     * @param defaultBlobVersion the blobVersion to use by default
+     * @return SecretKey Get Secret Key based on API level to use in encryption.
+     * @throws GeneralSecurityException
+     * @throws IOException
+     */
+    synchronized SecretKey loadSecretKeyForEncryption(String defaultBlobVersion) throws IOException,
```

Comment:
I think it's appropriate since it's the same with the existing "loadSecretKeyForEncryption()" , which takes no arguments. "loadSecretKeyForEncryption()" is very difficult to test, and thus raised issue like https://github.com/AzureAD/azure-activedirectory-library-for-android/issues/580  , it's suppressed now and the code coverage is lowered. Now I am adding another method with only one more argument in the method declaration, and it's the real implementation of "loadSecretKeyForEncryption()" . So now it's testable and code coverage is better.